storage/interceptors/broker_notifications_interceptor.go

```diff
@@ -164,19 +158,79 @@ func fetchBrokerPlans(ctx context.Context, brokerID string, repository storage.R
 	return objList.(*types.ServicePlans).ServicePlans, nil
 }
 
-func getSupportedPlatformsForPlans(plans []*types.ServicePlan) []string {
-	platformTypes := make(map[string]bool)
+func ResolveSupportedPlatformIDsForPlans(ctx context.Context, plans []*types.ServicePlan, repository storage.Repository) ([]string, error) {
+	var platformTypes map[string]bool
+	platformNames := make(map[string]bool)
 	for _, plan := range plans {
-		types := plan.SupportedPlatforms()
-		for _, t := range types {
-			platformTypes[t] = true
+		planSupportedPlatformNames := plan.SupportedPlatformNames()
+		if planSupportedPlatformNames == nil {
+			// no explicit supported platform names defined - collect the supported platform types
+			if platformTypes == nil {
+				//only initialize this map if any plan not specifying explicit platform names is found
+				platformTypes = make(map[string]bool)
+			}
+
```

Comment:
This case is flawed. Imagine the scenario where a plan has neither supported platform names, nor supported platform types -> result is the `platformType` map will get initialized.

The next plan however has only support platform type `kubernetes` -> result is the platformTypes will now contain on element. As a result, the below code will only fetch kubernetes platforms instead of all platforms (due to the previous plan).

IMO, the loop should terminate at the first occurrence when a plan with no supported types/names is encountered, no further processing should be made other than fetching all platform IDs.

Also as my previous comments mentions we should get rid of these nil checks as they cause unforeseen/hidden bugs like the one that was about to happen here.

Reply:
good catch! fixed (i think)